src/gui/folderman.cpp

```diff
@@ -1631,4 +1651,32 @@ void FolderMan::restartApplication()
     }
 }
 
+void FolderMan::slotReconnectToPushNotificationsForFiles(const Folder::Map &folderMap)
+{
+    // Disconnect the signal handlers
+    for (auto folder : folderMap) {
```

Comment:
Although I'm usually not a huge fan of it, I guess this could all be simplified using Qt::UniqueConnection.

https://doc.qt.io/qt-5/qt.html#ConnectionType-enum